src/View/Helper/FlashHelper.php

```diff
@@ -77,13 +89,54 @@ public function render($key = 'flash', array $options = [])
                 $key
             ));
         }
+
+        if (is_numeric(key($flash))) {
+            $this->request->session()->delete("Flash.$key");
+            return $this->_renderStack($flash, $options);
+        }
+
         $flash = $options + $flash;
         $this->request->session()->delete("Flash.$key");
+        return $this->_render($flash);
+    }
 
+    /**
+     * Renders a single message by calling its element
+     *
+     * @param array $flash Flash message parameter
+     * @return string|void Rendered flash message or null if flash key does not exist
+     *   in session.
+     */
+    protected function _render(array $flash)
+    {
         return $this->_View->element($flash['element'], $flash);
     }
 
     /**
+     * Renders the given stack of messages
+     *
+     * @param array $messages Messages to render
+     * @param array $options Additional options to use for the creation of this flash message.
+     *    Supports the 'params', and 'element' keys that are used in the helper.
+     * @return string The full stack rendered as a string
+     */
+    protected function _renderStack(array $messages, array $options = [])
+    {
+        $out = '';
+        foreach ($messages as $message) {
+            $message = $options + $message;
```

Comment:
Well actually, it was to follow the original options attribution in the ``render()`` method : https://github.com/cakephp/cakephp/blob/3.0/src/View/Helper/FlashHelper.php#L80

Since ``$options`` is the one passed to the render method, I figured it should be the same when the stack is involved, to have consistency in the rendering behavior.